Libraries/LibWeb/Worker/WebWorkerServer.ipc

```diff
@@ -6,7 +6,13 @@
 
 endpoint WebWorkerServer {
 
-    start_dedicated_worker(URL::URL url, Web::Bindings::WorkerType type, Web::Bindings::RequestCredentials credentials, String name, Web::HTML::TransferDataHolder message_port, Web::HTML::SerializedEnvironmentSettingsObject outside_settings) =|
+    start_worker(URL::URL url,
+                 Web::Bindings::WorkerType type,
+                 Web::Bindings::RequestCredentials credentials,
+                 String name,
+                 Web::HTML::TransferDataHolder message_port,
+                 Web::HTML::SerializedEnvironmentSettingsObject outside_settings,
+                 bool is_shared) =|
```

Comment:
Would prefer a named enum here, but won't fight too strongly on that since this matches spec text

Reply:
Yeah, I figure once we get an actual service worker entity going in this, we'll just swap it out for an enum WorkerType { Dedicated, Shared, Service, AudioWorklet, ... } or something.

Reply:
I'm being slow at working on it, but I was planning on a different function / IPC call altogether for it since the agorithm is in a different spec

But we already do have AgentType enum that could be used

Reply:
I'm going to go ahead and use AgentType after moving it to its own header